src/Network/Connection.php

```diff
@@ -336,7 +352,7 @@ public function writeFrame(Frame $stompFrame)
             throw new ConnectionException('Not connected to any server.', $this->activeHost);
         }
         $data = $stompFrame->__toString();
-        if (!@fwrite($this->connection, $data, strlen($data))) {
+        if (@fwrite($this->connection, $data, strlen($data)) !== strlen($data)) {
```

Comment:
IIRC this is not a good idea on non-blocking streams.
We either need to buffer/remember what was not yet written, or do a retry in case we have no buffer